tuxemon/event/actions/translated_dialog.py

```diff
@@ -25,68 +24,54 @@ class TranslatedDialogAction(EventAction):
     translation key. Parameters passed to the translation string will also
     be checked if a translation key exists.
 
-    You may also use special variables in dialog events. Here is a list
-    of available variables:
-
-    * ${{name}} - The current player's name.
-
-    Parameters following the translation name may represent one of two things:
-    If a parameter is var1=value1, it represents a value replacement.
-    If it's a single value (an integer or a string), it will be used as an
-    avatar image.
-    TODO: This is a hack and should be fixed later on, ideally without
-    overloading the parameters.
-
     Script usage:
         .. code-block::
 
-            translated_dialog <text>,<avatar>
-            translated_dialog <text>[,var1=value1]...
+            translated_dialog <text>,<avatar>,<bg>,<style>
 
     Script parameters:
         text: Text of the dialog.
         avatar: Monster avatar. If it is a number, the monster is the
             corresponding monster slot in the player's party.
             If it is a string, we're referring to a monster by name.
+        style: a predefined style in db/dialogue/dialogue.json
 
     """
 
     name = "translated_dialog"
-    raw_parameters: Sequence[str] = field(init=False)
-
-    def __init__(self, *args: Any) -> None:
-        super().__init__()
-        self.raw_parameters = args
+    raw_parameters: str
+    avatar: Union[int, str, None] = None
+    style: Optional[str] = None
 
     def start(self) -> None:
-        key = self.raw_parameters[0]
-        replace = []
+        key = process_translate_text(self.session, self.raw_parameters, [])
+
         avatar = None
-        for param in self.raw_parameters[1:]:
-            if "=" in param:
-                replace.append(param)
-            else:
-                avatar = get_avatar(self.session, param)
-
-        self.open_dialog(
-            process_translate_text(
-                self.session,
-                key,
-                replace,
-            ),
-            avatar,
+        if self.avatar:
+            avatar = get_avatar(self.session, self.avatar)
+
+        dialogue = "default"
+        if self.style:
+            dialogue = self.style
+
+        style = db.lookup(dialogue, table="dialogue")
+
+        colors: dict[str, Any] = {
+            "bg_color": string_to_colorlike(style.bg_color),
+            "font_color": string_to_colorlike(style.font_color),
+            "font_shadow": string_to_colorlike(style.font_shadow_color),
+            "border": style.border_path,
+        }
+
+        open_dialog(
+            session=self.session,
+            text=key,
```

Comment:
i think `key` should be `process_translate_text(..., key...)`?

Reply:
sure, this will be updated as soon as the other gets merged

Reply:
it's done above
`key = process_translate_text(self.session, self.raw_parameters, [])`